### PR TITLE
PT-6925: Support for Apps inside modules

### PR DESCRIPTION
**Pull request by OlegoO**
## Description
feat: Add support for Apps inside VC modules.

### CODE REVIEW

File: src/VirtoCommerce.Platform.Core/Modularity/ManifestModuleInfo.cs [hunk 1]
```diff
@@ -90,6 +93,12 @@ public virtual ManifestModuleInfo LoadFromManifest(ModuleManifest manifest)
             {
                 Groups.AddRange(manifest.Groups);
             }
+
+            if(manifest.Apps!=null)
```
Comment (ksavosteev) [inline src/VirtoCommerce.Platform.Core/Modularity/ManifestModuleInfo.cs:97]: fix formatting

File: src/VirtoCommerce.Platform.Web/Controllers/Api/AppsController.cs [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+        {
+            var retVal = _localModuleCatalog.Modules.OfType<ManifestModuleInfo>()
+                .SelectMany(x => x.Apps)
+                .Where(x => string.IsNullOrEmpty(x.Permission) || User.HasGlobalPermission(x.Permission))
+                .OrderBy(x => x.Title)
+                .Select(x => new AppDescriptor(x))
+                .ToArray();
+
+            return Ok(retVal);
+
```
Comment (ksavosteev) [inline src/VirtoCommerce.Platform.Web/Controllers/Api/AppsController.cs:36]: empty line

File: src/VirtoCommerce.Platform.Web/Controllers/Api/AppsController.cs [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+    {
+        private readonly ILocalModuleCatalog _localModuleCatalog;
+
+        public AppsController(ILocalModuleCatalog externalModuleCatalog)
+        {
+            _localModuleCatalog = externalModuleCatalog;
+        }
+
+        [HttpGet]
+        public ActionResult<AppDescriptor[]> GetApps(bool checkPermissions = false)
```
[the remark this answers is not in the document]
Comment (tatarincev) [reply src/VirtoCommerce.Platform.Web/Controllers/Api/AppsController.cs:26]: It is better imho to limit authorization check only to UI

File: src/VirtoCommerce.Platform.Web/Controllers/Api/AppsController.cs [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+using static OpenIddict.Abstractions.OpenIddictConstants;
+
+namespace VirtoCommerce.Platform.Web.Controllers.Api
+{
+    [Route("api/platform/apps")]
+    public class AppsController : Controller
+    {
+        private readonly ILocalModuleCatalog _localModuleCatalog;
+
+        public AppsController(ILocalModuleCatalog externalModuleCatalog)
```
[the remark this answers is not in the document]
Comment (OlegoO) [reply src/VirtoCommerce.Platform.Web/Controllers/Api/AppsController.cs:20]: > Are you sure what you want to use this icon platform_app.svg for platform APP?
No. Just for current preview
externalModuleCatalog -> localModuleCatalog fixed

File: src/VirtoCommerce.Platform.Web/Controllers/Api/AppsController.cs (hunk 3 above)
[the remark this answers is not in the document]
Comment (OlegoO) [reply src/VirtoCommerce.Platform.Web/Controllers/Api/AppsController.cs:26]: Fixed

**Pull request merged**